src/Microsoft.Extensions.Configuration.AzureAppConfiguration/IConfigurationRefresher.cs

```diff
@@ -17,6 +18,11 @@ public interface IConfigurationRefresher
         /// </summary>
         Uri AppConfigurationEndpoint { get; }
 
+        /// <summary>
+        /// An <see cref="ILoggerFactory"/> for creating a logger to log errors.
+        /// </summary>
+        ILoggerFactory LoggerFactory { get; set; }
```

Comment:
To avoid introducing this new property (a breaking change), we can cast the `IConfigurationRefresher` to `AzureAppConfigurationRefresher` in the refresherProvider.

Reply:
Do you mean cast `IConfigurationRefresher` to `AzureAppConfigurationProvider`? `AzureAppConfigurationRefresherProvider`  loops over all `IConfigurationProvider`s . So we cannot cast it to `AzureAppConfigurationRefresher`.

Reply:
This is possible, but I'm not sure if its the best thing to do. 
Are we saying that `AzureAppConfigurationRefresherProvider` needs to know about a concrete type (`AzureAppConfigurationProvider`) in addition to an interface (`IConfigurationRefresher`)?

Reply:
It's not a problem for one concrete type to know the other concrete type.

However, you brought up a good point in the other comment. This can be useful for scenarios DI is not used. So I'm good to keep this.